shared/chroot.go

```diff
@@ -89,8 +89,31 @@ func moveMounts(mounts []ChrootMount) error {
 			target = newTarget
 		}
 
+		// If the target's parent directory is a symlink, we need to resolve that as well.
+		targetDir := filepath.Dir(target)
+		for {
```

Comment:
I don't see what the for loop is for, unless the resolution fails, you'll never get a symlink twice in a row.